source/hpack/tables.d

```diff
@@ -0,0 +1,340 @@
+		10:	   HTTP2HeaderTableField(":status", HTTPStatus.partialContent), 		// 206
+		11:	   HTTP2HeaderTableField(":status", HTTPStatus.notModified), 			// 304
+		12:	   HTTP2HeaderTableField(":status", HTTPStatus.badRequest), 			// 400
+		13:	   HTTP2HeaderTableField(":status", HTTPStatus.notFound), 				// 404
+		14:	   HTTP2HeaderTableField(":status", HTTPStatus.internalServerError), 	// 500
+		15:	   HTTP2HeaderTableField("accept-charset", ""),
+		16:	   HTTP2HeaderTableField("accept-encoding", ["gzip", "deflate"]),
+		17:	   HTTP2HeaderTableField("accept-language", ""),
+		18:	   HTTP2HeaderTableField("accept-ranges", ""),
+		19:	   HTTP2HeaderTableField("accept", ""),
+		20:	   HTTP2HeaderTableField("access-control-allow-origin", ""),
+		21:	   HTTP2HeaderTableField("age", ""),
+		22:	   HTTP2HeaderTableField("allow", ""),
+		23:	   HTTP2HeaderTableField("authorization", ""),
+		24:	   HTTP2HeaderTableField("cache-control", ""),
+		25:	   HTTP2HeaderTableField("content-disposition", ""),
+		26:	   HTTP2HeaderTableField("content-encoding", ""),
+		27:	   HTTP2HeaderTableField("content-language", ""),
+		28:	   HTTP2HeaderTableField("content-length", ""),
+		29:	   HTTP2HeaderTableField("content-location", ""),
+		30:	   HTTP2HeaderTableField("content-range", ""),
+		31:	   HTTP2HeaderTableField("content-type", ""),
+		32:	   HTTP2HeaderTableField("cookie", ""),
+		33:	   HTTP2HeaderTableField("date", ""),
+		34:	   HTTP2HeaderTableField("etag", ""),
+		35:	   HTTP2HeaderTableField("expect", ""),
+		36:	   HTTP2HeaderTableField("expires", ""),
+		37:	   HTTP2HeaderTableField("from", ""),
+		38:	   HTTP2HeaderTableField("host", ""),
+		39:	   HTTP2HeaderTableField("if-match", ""),
+		40:	   HTTP2HeaderTableField("if-modified-since", ""),
+		41:	   HTTP2HeaderTableField("if-none-match", ""),
+		42:	   HTTP2HeaderTableField("if-range", ""),
+		43:	   HTTP2HeaderTableField("if-unmodified-since", ""),
+		44:	   HTTP2HeaderTableField("last-modified", ""),
+		45:	   HTTP2HeaderTableField("link", ""),
+		46:	   HTTP2HeaderTableField("location", ""),
+		47:	   HTTP2HeaderTableField("max-forwards", ""),
+		48:	   HTTP2HeaderTableField("proxy-authenticate", ""),
+		49:	   HTTP2HeaderTableField("proxy-authorization", ""),
+		50:	   HTTP2HeaderTableField("range", ""),
+		51:	   HTTP2HeaderTableField("referer", ""),
+		52:	   HTTP2HeaderTableField("refresh", ""),
+		53:	   HTTP2HeaderTableField("retry-after", ""),
+		54:	   HTTP2HeaderTableField("server", ""),
+		55:	   HTTP2HeaderTableField("set-cookie", ""),
+		56:	   HTTP2HeaderTableField("strict-transport-security", ""),
+		57:	   HTTP2HeaderTableField("transfer-encoding", ""),
+		58:	   HTTP2HeaderTableField("user-agent", ""),
+		59:	   HTTP2HeaderTableField("vary", ""),
+		60:	   HTTP2HeaderTableField("via", ""),
+		61:	   HTTP2HeaderTableField("www-authenticate", "")
+	];
+}
+
+private struct DynamicTable {
+	private {
+		// table is a queue, initially empty
+		DList!HTTP2HeaderTableField m_table;
```

Comment:
Instead of a `DList`, a circular buffer seems more suitable here. There is `vibe.internal.array.FixedRingBuffer` which can be used here. This will make `opIndex` an O(1) operation and avoids the per-element memory allocation when inserting.